sdk/src/main/scala/io/horizen/account/state/nativescdata/forgerstakev2/DelegateCmdInputDecoder.scala

```diff
@@ -0,0 +1,47 @@
+package io.horizen.account.state.nativescdata.forgerstakev2
+
+import io.horizen.account.abi.{ABIDecoder, ABIEncodable, MsgProcessorInputDecoder}
+import io.horizen.account.state.{AddNewStakeCmdInput, ForgerPublicKeys}
+import io.horizen.evm.Address
+import io.horizen.proposition.{PublicKey25519Proposition, VrfPublicKey}
+import org.web3j.abi.TypeReference
+import org.web3j.abi.datatypes.generated.{Bytes1, Bytes32}
+import org.web3j.abi.datatypes.{StaticStruct, Type, Address => AbiAddress}
+
+import java.util
+
+object DelegateCmdInputDecoder
+  extends ABIDecoder[DelegateCmdInput]
+    with MsgProcessorInputDecoder[DelegateCmdInput] {
+
+  override val getListOfABIParamTypes: util.List[TypeReference[Type[_]]] =
+    org.web3j.abi.Utils.convert(util.Arrays.asList(
+      new TypeReference[Bytes32]() {},
+      new TypeReference[Bytes32]() {},
+      new TypeReference[Bytes1]() {}
+      ))
+
+  override def createType(listOfParams: util.List[Type[_]]): DelegateCmdInput = {
+    val forgerPublicKey = new PublicKey25519Proposition(listOfParams.get(0).asInstanceOf[Bytes32].getValue)
+    val vrfKey = decodeVrfKey(listOfParams.get(1).asInstanceOf[Bytes32], listOfParams.get(2).asInstanceOf[Bytes1])
+    val forgerPublicKeys = ForgerPublicKeys(forgerPublicKey, vrfKey)
+    DelegateCmdInput(forgerPublicKeys)
+  }
+
+  private[horizen] def decodeVrfKey(vrfFirst32Bytes: Bytes32, vrfLastByte: Bytes1): VrfPublicKey = {
```

Comment:
There are multiple definition of this same function, consider moving it in a proper common place